plugins/SegmentEditor/API.php

```diff
@@ -395,11 +395,37 @@ public function getAll($idSite = false)
             }
         }
 
+        $segments = $this->filterSegmentsWithDisabledElements($segments, $idSite);
         $segments = $this->sortSegmentsCreatedByUserFirst($segments);
 
         return $segments;
     }
 
+    /**
+     * Filter out any segments which cannot be initialized due to disable plugins or features
+     *
+     * @param array $segments
+     * @param       $idSite
+     *
+     * @return array
+     */
+    private function filterSegmentsWithDisabledElements(array $segments, $idSite = false): array
+    {
+        $sites = [];
+        if (!empty($idSite)) {
+            $sites = [$idSite];
```

Comment:
Would something like this be needed here or idSite may never be provided as an array?

```suggestion
            $sites = is_array($idSite) ? $idSite : [$idSite];
```

Reply:
The parent parameter documentation for idSite says bool|int, but it's not actually typed so it's possible that an array could be passed, might be good to be defensive here :+1: